### ISSUE TITLE
WebAccess: ENOENT: Out of sync!

### ISSUE THREAD

**Issue by yoursunny** [zen-fs/dom#23]
@zenfs/dom 1.1.0 is incompatible with @zenfs/core 1.6.11.

## package.json

```json
{
  "private": true,
  "type": "module",
  "packageManager": "pnpm@9.15.1+sha512.1acb565e6193efbebda772702950469150cf12bcc764262e7587e71d19dc98a423dff9536e57ea44c49bdf790ff694e83c27be5faa23d67e0c033b583be4bfcf",
  "dependencies": {
    "@types/node": "^22.10.2",
    "@zenfs/core": "^1.6.11",
    "@zenfs/dom": "^1.1.0",
    "typescript": "~5.7.2",
    "webpack": "^5.97.1",
    "webpack-cli": "^6.0.1",
    "webpack-dev-server": "^5.2.0"
  }
}
```

## webpack.config.cjs

```js
const path = require("node:path");

/** @return {import("webpack").Configuration} */
module.exports = (env, argv) => ({
  mode: argv.mode ?? "production",
  devtool: argv.mode === "development" ? "cheap-module-source-map" : "source-map",
  entry: "./src/main.js",
  performance: {
    maxAssetSize: Infinity,
    maxEntrypointSize: Infinity,
  },
  output: {
    filename: "bundle.js",
    path: path.resolve(__dirname, "public"),
  },
  node: false,
  devServer: {
    allowedHosts: "all",
    port: 3333,
    static: {
      directory: path.resolve(__dirname, "public"),
    },
  },
});
```

## src/main.js

```js
import { configureSingle, fs } from "@zenfs/core";
import { WebAccess } from "@zenfs/dom";

document.querySelector("#button").addEventListener("click", async () => {
  const root = await navigator.storage.getDirectory();
  await configureSingle({ backend: WebAccess, handle: root });
  console.log(await fs.promises.readdir("/"));
  await fs.promises.writeFile("/A.bin", Uint8Array.of(0xC0, 0xC1));
  console.log(await fs.promises.readdir("/"));
});
```

## public/index.html

```html
<!DOCTYPE html>
<meta charset="utf-8">
<title>zenfs-demo</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<div><input id="button" type="button" value="activate"/></div>
<script src="bundle.js"></script>
```

## Command Line and Manual Action

```bash
./node_modules/.bin/webpack
python3 -m http.server -d ./public
```

Open `http://localhost:8000` in Firefox 129.0.
Open DevTools and switch to **Console** tab.
Click **activate** button.

## Error Message

```text
Uncaught (in promise) Error: ENOENT: Out of sync! (ENOENT: No such file or directory, '/A.bin'), '/A.bin'
    a error.js:259
    e async.js:255
    crossCopy async.js:185
    crossCopy async.js:212
    ready async.js:108
    ki config.js:50
    async* config.js:60
    <anonymous> main.js:6
    async* main.js:4
    <anonymous> main.js:9
    <anonymous> main.js:9
```

## Related Inquiry

I only use `fs.promises` API but do not need any synchronous operations.
How to disable the sync cache and prefetching completely?

**Comment by james-pre** [comment]
Hi @yoursunny, you can pass the `disableAsyncCache` config option.

It should be compatible, I'll take a look. Thanks for opening an issue.

**Comment by james-pre** [comment]
@yoursunny,

This should be fixed in `@zenfs/core` [v1.6.12](https://github.com/zen-fs/core/releases/tag/v1.6.12). Please feel free to reach out if you have any questions or issues.
